src/target/riscv/riscv-013.c

```diff
@@ -595,6 +596,7 @@ static int dmi_op_timeout(struct target *target, uint32_t *data_in,
 
 	if (status != DMI_STATUS_SUCCESS) {
 		LOG_ERROR("Failed %s at 0x%x; status=%d", op_name, address, status);
+		dtmcontrol_scan(target, DTM_DTMCS_DMIRESET);
 		return ERROR_FAIL;
 	}
```

Comment:
No need to call `dtmcontrol_scan(target, DTM_DTMCS_DMIRESET);` here. It has already been handled above.

The word "DMI" can be included in the error message to make it clearer to the user what type of error it actually is.

```suggestion
	if (status != DMI_STATUS_SUCCESS) {
		LOG_ERROR("Failed DMI %s at 0x%x; status=%d", op_name, address, status);
		return ERROR_FAIL;
	}
```

Reply:
Done.